routes/items.js

```diff
@@ -15,6 +16,15 @@ router.get( '/', ( request, response ) => {
     .then( respondWithItems( user, data => response.render( 'items/index', data )))
 })
 
+router.get( '/starred', ( request, response ) => {
+  const Item = request.app.get( 'models' ).Item
+  const { user, query } = request
+
+  buildStarredItemArray( Item, user.id )
+    .then( data => response.json( { data } ) )
+    .then( respondWithStarred(user, data => response.render( 'items/starred', data )))
```

Comment:
You can't respond twice - you probably need to remove this `then`, especially since this endpoint is intended to return json that will be handled on the client, rather than render a view.